### PR TITLE
Improve performance of fast test by reducing down blocks

### LINKED ISSUE #5260: [HacktoberFest] make fast test for `StableDiffusionXLPipeline` smaller when applicable
This feature request is included in this year's Diffusers Hacktoberfest 🎃! 

Hacktoberfest's purpose is to give upcoming open-source developers the opportunity to learn about open-source and programming by fixing simple issues (like this one).

**How to solve the issue**:

- 1. Every issue has a description further below explaining the problem. First, you should try to understand the issue.
- 2. Second, follow the [contribution guide](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md) to learn how to contribute to Diffusers.\
- 3. Claim this issue for you by opening a PR that links to this issue and writing, "I am solving this issue in `<link-to-pr>`".

**Issue**

We want to run our fast tests run faster. But what is a fast test for a pipeline, though? 

This class holds all fast tests here: https://github.com/huggingface/diffusers/blob/bbe8d3ae1330ecc1b90a9873b60d290a83254989/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L44 

The idea of this issue is to try to identify if the pipeline components (such as the UNet, text encoders, VAE) could be made smaller without disturbing the correctness of the tests. 

For example, could we reduce the number of blocks in the UNet? This is how the UNet is initialized:

https://github.com/huggingface/diffusers/blob/bbe8d3ae1330ecc1b90a9873b60d290a83254989/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L53

Similar idea could be applied to the [VAE](https://github.com/huggingface/diffusers/blob/bbe8d3ae1330ecc1b90a9873b60d290a83254989/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L78), and text encoders ([one](https://github.com/huggingface/diffusers/blob/bbe8d3ae1330ecc1b90a9873b60d290a83254989/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L88), [two](https://github.com/huggingface/diffusers/blob/4f74a5e1f7d9274931f8d782864f769266a24ecf/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L105)). 

As a consequence of making the pipeline components smaller, the expected assertion values ([example](https://github.com/huggingface/diffusers/blob/bbe8d3ae1330ecc1b90a9873b60d290a83254989/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L147)) might change but it should not impact the correctness of the test, whatsoever. 

Cc: @DN6

### PR DESCRIPTION
**Pull request by sepal**
# What does this PR do?
Fixes #5260 by reducing the channels of the downblocks of the UNet model.

So far all tests pass on my local machine without any changes, except for `test_stable_diffusion_xl_euler`. The deviation for image slice check is bigger than 0.01, which is why I changed the assert to 0.1. Are we ok with this? 

According to `time` the running the tests without the optimizations took 2:13.44, with it only took 1:04.99 total, so the speed up seems worth it.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sepal** [comment]
@sayakpaul can I add you as a reviewer?

File: tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py [hunk 1]
```diff
@@ -146,7 +147,7 @@ def test_stable_diffusion_xl_euler(self):
         assert image.shape == (1, 64, 64, 3)
         expected_slice = np.array([0.5873, 0.6128, 0.4797, 0.5122, 0.5674, 0.4639, 0.5227, 0.5149, 0.4747])
```
Comment (patrickvonplaten) [inline tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py:148]: We'll have to update the values here I'm afraid :-) Good idea to change the number of channels

Comment (sepal) [reply tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py:148]: How did the team come with the numbers in the first place?

Comment (sepal) [reply tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py:148]: haha, ok. I thought, that maybe calculated the values *manually*, but if just using what I get is an option, then I'll do that :-)

**Review by DN6 — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**